### ISSUE TITLE
perf: make dispatcher loop stats work with statsd

### ISSUE THREAD

**Issue by mattklein123** [envoyproxy/envoy#8109]
I've been debugging a perf problem at Lyft and the dispatcher loop stats are indeed incredibly useful. We should make these stats usable with statsd. The best way I can see to do this is to potentially have the ability to tag a histogram as "high rate" so that it gets ignored by statsd for individual flushes, and then statsd could potentially flush high rate histograms as gauges with various percentiles dumped?

**Comment by mattklein123** [comment]
@mergeconflict @jmarantz any thoughts on this?

**Comment by mergeconflict** [comment]
Yeah, I think exporting gauges for each entry in `HistogramStatistics::supportedQuantiles` is a fine workaround.

I can't really think of a design that I'm happy with for tagging histograms as "high rate," though. If you added it to the `Histogram` interface, it'd only be meaningful to the statsd sink, and I don't love that. Maybe I'd like it better if we call the histogram-to-gauges transformation something else (more a statement of what it is, rather than what problem it's solving) and apply it regardless of which stats sink is in use. I'm bad at naming things, so my suggestion is `convertToGauges` :)

**Comment by mattklein123** [comment]
Yeah I agree it's not great. Let me come up with a more concrete proposal when I come back to this. If anyone else wants to work on this in the meantime let me know.
